packages/itwin/measure-tools/src/api/DrawingTypeDataCache.ts

```diff
@@ -0,0 +1,39 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import type { IModelConnection } from "@itwin/core-frontend";
+import { IModelApp } from "@itwin/core-frontend";
+import { SheetMeasurementsHelper } from "./SheetMeasurementHelper";
+
+export class DrawingDataCache {
+
+  private _drawingTypeCache: SheetMeasurementsHelper.DrawingTypeData[];
+  private _sheetChangeListener: () => void = () => {};
+
+  public get drawingtypes() {
+    return this._drawingTypeCache;
+  }
+
+  public constructor() {
+    this._drawingTypeCache = [];
+  }
+
+  public async updateDrawingTypeCache(iModel: IModelConnection) {
+    this._sheetChangeListener();
+    const sheetIds = [];
+
+    for (const viewport of IModelApp.viewManager) {
+      if (viewport.view.isSheetView()) {
```

Comment:
There could be more than one sheet view open at any given time. I would suggest changing to an array of listeners, namely `_sheetChangeListener: VoidFunction[] = []`